public/apps/configuration/panels/audit-logging/edit-setting-group.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { Fragment } from 'react';
+import {
+  EuiCodeBlock,
+  EuiComboBox,
+  EuiDescribedFormGroup,
+  EuiFormRow,
+  EuiSpacer,
+  EuiSwitch,
+  EuiTitle,
+} from '@elastic/eui';
+import { get } from 'lodash';
+import { displayBoolean } from '../../utils/display-utils';
+import { comboBoxOptionToString, stringToComboBoxOption } from '../../utils/combo-box-utils';
+
+export function EditSettingGroup(props: {
+  settingGroup: object;
+  config: object;
+  handleChange: (setting: object, val: string) => void;
+  handleInvalid: (setting: object, val: string) => void;
```

Comment:
Please have type definitions for these objects.

Reply:
"settingGroup" is actually a constant object, e.g
  LAYER_SETTINGS: {
    title: CONFIG_LABELS.LAYER_SETTINGS,
    settings: [
      CONFIG.AUDIT.REST_LAYER,
      CONFIG.AUDIT.REST_DISABLED_CATEGORIES,
      CONFIG.AUDIT.TRANSPORT_LAYER,
      CONFIG.AUDIT.TRANSPORT_DISABLED_CATEGORIES,
    ],
  },

It is all consisted of static strings. Thus it is just a constant object.

"handleInvalid" is not used yet. Let me remove it.

Reply:
Even for constant I would suggest still add type definition, like 
https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/dev/public/apps/configuration/app-router.tsx#L33
https://github.com/opendistro-for-elasticsearch/security-kibana-plugin/blob/dev/public/apps/configuration/types.ts#L32-L35
So in the future when adding some more constants, no chance to mistype a field name.

Reply:
Checked the examples and used similar type definitions. Thanks.